src/utils/decorators/jsTsDecorators.ts

```diff
@@ -0,0 +1,304 @@
+import path from "path";
+import FileDetails from "../../models/fileDetails.js";
+import { DecoratorExtractor } from "./decoratorFactory.js";
+import fs from "fs";
+import { ClassInfo, MethodInfo } from "./decoratorTypes.js";
+import { NodePath } from "@babel/traverse";
+import {
+    ClassDeclaration,
+    ClassMethod,
+    isCallExpression,
+    isIdentifier,
+    isClassDeclaration,
+    CallExpression,
+    isObjectExpression,
+    isObjectProperty,
+    isStringLiteral,
+} from "@babel/types";
+import babel from "@babel/core";
+import { createRequire } from "module";
+import { default as Parser } from "tree-sitter";
+
+export class JsTsDecoratorExtractor implements DecoratorExtractor {
+    fileFilter(cwd: string): (file: FileDetails) => boolean {
+        return (file: FileDetails) => {
+            const folderPath = path.join(cwd, file.path);
+            return (
+                (file.extension === ".js" || file.extension === ".ts") &&
+                !file.path.includes("node_modules") &&
+                !file.path.includes(".git") &&
+                !fs.lstatSync(folderPath).isDirectory()
+            );
+        };
+    }
+
+    async getDecoratorsFromFile(file: string): Promise<ClassInfo[]> {
+        const inputCode = fs.readFileSync(file, "utf8");
```

Comment:
maybe make a private function from all this code and name it more specific: `getDecoratorsUsingBabel`. And then `getDecoratorsFromFile` will call the two methods.